plugin/js/parsers/ReaperscansParser.js

```diff
@@ -0,0 +1,106 @@
+"use strict";
+
+parserFactory.register("reaperscans.com", () => new ReaperscansParser());
+
+class ReaperscansParser extends Parser{
+    constructor() {
+        super();
+        this.seriesInfo = null;
+    }
+
+    async getChapterUrls(dom) {
+        if(this.seriesInfo === null) {
+            await this.extractSeriesInfo(dom)
+        }
+        const chaptersApiUrl = new URL('https://api.reaperscans.com/chapter/query');
+        chaptersApiUrl.searchParams.set('page', 1);
+        chaptersApiUrl.searchParams.set('perPage', this.seriesInfo.chaptersCount);
+        chaptersApiUrl.searchParams.set('query', '');
+        chaptersApiUrl.searchParams.set('order', 'asc');
+        chaptersApiUrl.searchParams.set('series_id', this.seriesInfo.seriesId);
+        const chaptersJson = (await HttpClient.fetchJson(chaptersApiUrl)).json;
+        const chapterUrls = chaptersJson.data.map(chapter => {
+            return {
+                sourceUrl: `https://reaperscans.com/series/${this.seriesInfo.seriesSlug}/${chapter.chapter_slug}`,
+                title: `${chapter.chapter_name} - ${chapter.chapter_title}`
+            };
+        });
+        return chapterUrls;
+    }
+
+    async extractSeriesInfo(dom) {
+        const baseUrl= this.getBaseUrl(dom);
+        const match = baseUrl.match(/reaperscans\.com\/series\/[^/]+/);
+        const apiUrl = `https://api.${match[0]}`;
+        const seriesJson = (await HttpClient.fetchJson(apiUrl)).json
+        this.seriesInfo = {
+            seriesId: seriesJson.id,
+            seriesName: seriesJson.title,
+            seriesSlug: seriesJson.series_slug,
+            chaptersCount: seriesJson.meta.chapters_count
+        }
+    }
+
+    findScriptContent(dom) {
+        const scripts = dom.querySelectorAll("script");
+        for (const script of scripts) {
+            if (script.firstChild && script.firstChild.nodeName === "#text" 
+                && script.textContent.includes("\\u003cp") //contains paragraph tags
+                && script.textContent.includes("series_id") === false //ignore other series info
+                ) {
+                return script.textContent;
+            }
+        }
+        return null;
+    }
+
+    findContent(dom) {
+        const scriptContent = this.findScriptContent(dom);
+        const content = scriptContent.substring(scriptContent.indexOf('\\u003cp'), scriptContent.lastIndexOf('\\u003e') + 6)
+        const decoded = this.decodeString(content);
+        const html = this.cleanHTML(decoded);
+        return html;
+    }
+
+    decodeString(input) {
+        // Decode Unicode escapes
+        const unicodeDecoded = JSON.parse(`"${input}"`);
+        // Decode HTML entities
+        const textarea = document.createElement("textarea");
+        textarea.innerHTML = unicodeDecoded;
```

Comment:
Google and Mozilla may refuse to accept an extension that assigns text to innerHTML.  You should use DOMParser's parseFromString() to convert string to HTML.  https://developer.mozilla.org/en-US/docs/Web/API/DOMParser/parseFromString